### PR TITLE
docs: add section about mocking to README

### LINKED ISSUE #133: Document mocking next-firebase-auth for test suites
**Did you have difficulty using or understanding an aspect of `next-firebase-auth`?**
The documentation does not currently cover examples of testing components/pages wrapped with `next-firebase-auth`. Conversation about this has been ongoing in discussion #124 for a couple of days. I think it would be helpful for other users to include some documentation about setting this up.

**What documentation would you like to add or change?**
I would like to see a new section in the documentation that describes how to mock the library, and how to use the mocks in tests.

**Version**
0.13.0-alpha0

### PR DESCRIPTION
**Pull request by grahaml**
Update README to include a new Test Examples section, including some information about how to mock `next-firebase-auth` in a `Jest` suite.

Closes #133

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -496,6 +496,204 @@ We expect some apps will need some features that are not currently available:
...
+// in the context of this test run. This allows you to manipulate the return value of each
+// function within this test suite.
+import { useAuthUser, withAuthUser } from 'next-firebase-auth';
+
+// Import your mock AuthUser generator
+import getMockAuthUser from '../../utils/test-utils/get-mock-auth-user';
+
+// Tell jest to mock the whole library, which will basically replace it with the default export
+// from `__mocks__/next-firebase-auth/`.
+jest.mock('next-firebase-auth');
```
Comment (kmjennison) [inline README.md:605]: If using [manual mocks for Node modules](https://jestjs.io/docs/manual-mocks#mocking-node-modules), they'll be automatically mocked.

File: README.md [hunk 2]
```diff
@@ -496,6 +496,204 @@ We expect some apps will need some features that are not currently available:
...
```
Comment (kmjennison) [inline README.md:660]: Same -- this is optional.

File: README.md [hunk 3]
```diff
@@ -496,6 +496,204 @@ We expect some apps will need some features that are not currently available:
 
 We'd love to hear your feedback on these or other features. Please feel free to [open a discussion](https://github.com/gladly-team/next-firebase-auth/discussions)!
 
+## Test examples
+
+### Jest
+
+In order to test components wrapped with functions from `next-firebase-auth`, you will need to mock the `next-firebase-auth` library. This can be achieved using the [manual mocks feature of Jest](https://jestjs.io/docs/manual-mocks).
+
+To start you need to stub out the entire module in a `__mocks__` folder next to `node_modules` in your application, eg.
```
Comment (kmjennison) [inline README.md:505]: I'm going to soften this language a bit, because there are other approaches someone could use to testing/mocking.

Comment (kmjennison) [reply README.md:505]: Will also note that the manual mocks are helpful for defining default behavior but not strictly necessary.

**Review by kmjennison — APPROVED** [review]
Thank you for the clear and thorough documentation! Will merge, make some small tweaks to language, and link to this from the top of the README.

**Comment by grahaml** [comment]
My pleasure! And of course please update as you see fit or let me know if I can help with any further changes :)

**Pull request merged**